chain/network/src/peer_manager/tests/sync_snapshot_hosts.rs

```diff
@@ -0,0 +1,282 @@
+fn take_sync_snapshot_msg(event: crate::peer::testonly::Event) -> Option<SyncSnapshotHosts> {
+    match event {
+        peer::testonly::Event::Network(PME::MessageProcessed(
+            tcp::Tier::T2,
+            PeerMessage::SyncSnapshotHosts(msg),
+        )) => Some(msg),
+        _ => None,
+    }
+}
+
+/// Used to consume peer manager events until there's an event of type SyncSnapshotHosts
+fn take_sync_snapshot_msg_manager(
+    event: crate::peer_manager::testonly::Event,
+) -> Option<SyncSnapshotHosts> {
+    match event {
+        peer_manager::testonly::Event::PeerManager(PME::MessageProcessed(
+            tcp::Tier::T2,
+            PeerMessage::SyncSnapshotHosts(msg),
+        )) => Some(msg),
+        _ => None,
+    }
+}
+
+/// Each actix arbiter (in fact, the underlying tokio runtime) creates 4 file descriptors:
+/// 1. eventfd2()
+/// 2. epoll_create1()
+/// 3. fcntl() duplicating one end of some globally shared socketpair()
+/// 4. fcntl() duplicating epoll socket created in (2)
+/// This gives 5 file descriptors per PeerActor (4 + 1 TCP socket).
+const FDS_PER_PEER: usize = 5;
+
+/// Test that PeerManager broadcasts SnapshotHostInfo messages to all connected peers
+#[tokio::test]
+async fn broadcast() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+    let clock = clock.clock();
+    let clock = &clock;
+
+    let pm = peer_manager::testonly::start(
+        clock.clone(),
+        near_store::db::TestDB::new(),
+        chain.make_config(rng),
+        chain.clone(),
+    )
+    .await;
+
+    tracing::info!(target:"test", "Connect a peer, expect initial sync to be empty.");
+    let peer1_config = chain.make_config(rng);
+    let mut peer1 =
+        pm.start_inbound(chain.clone(), peer1_config.clone()).await.handshake(clock).await;
+    let empty_sync_msg = peer1.events.recv_until(take_sync_snapshot_msg).await;
+    assert_eq!(empty_sync_msg.hosts, vec![]);
+
+    tracing::info!(target:"test", "Connect two more peers.");
+    let peer2_config = chain.make_config(rng);
+    let mut peer2 =
+        pm.start_inbound(chain.clone(), peer2_config.clone()).await.handshake(clock).await;
+    let empty_sync_msg = peer2.events.recv_until(take_sync_snapshot_msg).await;
+    assert_eq!(empty_sync_msg.hosts, vec![]);
+
+    let mut peer3 =
+        pm.start_inbound(chain.clone(), chain.make_config(rng)).await.handshake(clock).await;
+    let empty_sync_msg = peer3.events.recv_until(take_sync_snapshot_msg).await;
+    assert_eq!(empty_sync_msg.hosts, vec![]);
+
+    tracing::info!(target:"test", "Send a SyncSnapshotHosts message from peer1, make sure that all peers receive it.");
+
+    let info1 =
+        make_snapshot_host_info(&peer1_config.node_id(), 123, vec![0, 1], &peer1_config.node_key);
```

Comment:
Since your tests aren't at all sensitive to the contents of SnapshotHostInfo, it may help readability to pass `rng` into `make_snapshot_host_info` and generate some random `epoch_height` and `shards` inside of it rather than in-lining arbitrary values here.